### DISCUSSION TITLE
Issue using Teams permissions

### DISCUSSION THREAD

**Discussion by rubabnaziri** [spatie/laravel-permission#2070] [Q&A]
I have been struggling with an issue for quite some time now. Tried multiple approaches. 
I have Roles, Permissions, Users, Groups. Groups is Team in Spatie terms.
Roles have Permissions. 
Users have Roles
Groups have Roles and Users. 
A user may have multiple groups and multiple roles.

When I try to assign a role to the user I get a SQLSTATE[23502]: Not null violation: 7 ERROR: null value in column "group_id" of relation "model_has_roles" violates not-null constraint DETAIL: Failing row contains (2, user, 2, null).

**Comment by erikn69** [comment]
Read documentation or other discussions, you forget the `setPermissionsTeamId`
